src/main/java/io/strimzi/kafka/bridge/SinkBridgeEndpoint.java

```diff
@@ -357,6 +357,7 @@ private void partitionsAssignmentAndSeek() {
             Optional<PartitionInfo> requestedPartitionInfo =
                     availablePartitions.stream()
                             .filter(p -> p.getTopic().equals(topicSubscription.getTopic()) &&
+                                    topicSubscription.getPartition() != null &&
```

Comment:
can you explain in which case this can be `null`?

Reply:
If we do subscribe to a topic, it is stored into `topicSubscriptions`. Subscription request body might be for example 
```
{
  "topics": [
    "topic1",
    "topic2"
  ]
}
```
If we do assign after subscription (what is wrong because of exclusivity), we iterate through all topic subscriptions in `topicSubscriptions` and we are trying to get partitions. And because the subscription body does not have any partitions info, it is null. Then just comparing null causes NPE.